client/extensions/auto-translator/src/widget/translation-dialog/form.tsx

```diff
@@ -145,22 +145,35 @@ export const getTranslationDialogFormValues = (
   workingArticle: IArticle,
   articleVersions: IArticle[]
 ): TranslationDialogFormProps => {
-  const translations = articleVersions
+  const writethrus = articleVersions
     // version 0 is the initial object (contains no metadata)
-    .filter((article) => article._current_version !== 0)
-    .reduce<TranslationDialogFormProps["translations"]>(
-      (translations, article) => {
-        const images = getImagesFormValues(article);
-        const translationEntry = getTranslationEntryFormValues(article, images);
+    .filter(
+      (article) => article._current_version !== 0 && article.anpa_take_key
+    );
 
-        Object.assign(translations, {
-          [`${article._current_version}`]: translationEntry,
-        });
+  const translations = writethrus.length
+    ? writethrus.reduce<TranslationDialogFormProps["translations"]>(
+        (translations, article) => {
+          const images = getImagesFormValues(article);
+          const translationEntry = getTranslationEntryFormValues(
+            article,
+            images
+          );
 
-        return translations;
-      },
-      {}
-    );
+          Object.assign(translations, {
```

Comment:
It seems like we’re mutating the accumulator here, I've read that this could lead to unexpected side effects.

What do you think about the following?

`const translations = writethrus.length > 0
  ? writethrus.reduce((acc, article) => {
      const images = getImagesFormValues(article);
      const translationEntry = getTranslationEntryFormValues(article, images);

      return {
        ...acc,
        [article.anpa_take_key]: translationEntry,
      };
    }, {})
  : {
      current: getTranslationEntryFormValues(
        workingArticle,
        getImagesFormValues(workingArticle)
      ),
    };`

Reply:
there's no need to avoid mutating in reduce since I own the object. spreading also creates a new object, which is not necessary here